### PR TITLE
Auth/session management deprecations - part 5

### LINKED ISSUE #9714: Auth/session management deprecations - part 5
Ticket created based on #9622
That one contained a lot of work and I splitted the part that is not tightly related to auth module, but changed with the `epic/auth` merge.

- [x] `projects/core/src/cart/services/multi-cart-state-persistence.service.ts` - changed public functions
- [x] `projects/core/src/product/facade/product-reference.service.ts` - changed properties
- [x] `projects/core/src/global-message/http-interceptors/handlers/unauthorized/unauthorized.handler.ts` - removed
- [x] `projects/core/src/global-message/http-interceptors/handlers/forbidden/forbidden.handler.ts` - added constructor
- [x] `projects/core/src/checkout/facade/checkout-cost-center.service.ts` - changed constructor
- [x] `projects/core/src/checkout/facade/checkout-payment.service.ts` - changed constructor
- [x] `projects/core/src/checkout/facade/checkout.service.ts` - changed constructor
- [x] `projects/core/src/occ/adapters/checkout/converters/occ-replenishment-order-form-serializer.ts` - changed constructor
- [x] `projects/core/src/user/facade/customer-coupon.service.ts` - changed constructor
- [x] `projects/core/src/user/facade/order-return-request.service.ts` - changed constructor
- [x] `projects/core/src/user/facade/user-address.service.ts` - changed constructor
- [x] `projects/core/src/user/facade/user-consent.service.ts` - changed constructor
- [x] `projects/core/src/user/facade/user-interests.service.ts` - changed constructor
- [x] `projects/core/src/user/facade/user-notification-preference.service.ts` - changed constructor

### PR DESCRIPTION
**Pull request by psmul**
Closes: #9714

### CODE REVIEW

File: projects/schematics/src/migrations/3_0/methods-and-properties-deprecations/data/multi-cart-state-persistence.service.migration.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+} from '../../../../shared/constants';
+import { MethodPropertyDeprecation } from '../../../../shared/utils/file-utils';
+
+// projects/core/src/cart/services/multi-cart-state-persistence.service.ts
+export const MULTI_CART_STATE_PERSISTENCE_SERVICE_MIGRATION: MethodPropertyDeprecation[] = [
+  {
+    class: MULTI_CART_STATE_PERSISTENCE_SERVICE,
+    importPath: SPARTACUS_CORE,
+    deprecatedNode: SYNC,
+    comment: `// ${TODO_SPARTACUS} Method '${SYNC}' was removed from '${MULTI_CART_STATE_PERSISTENCE_SERVICE}'. Use ${INIT_SYNC} instead.`,
```
Comment (marlass) [inline projects/schematics/src/migrations/3_0/methods-and-properties-deprecations/data/multi-cart-state-persistence.service.migration.ts]: Let's just say that it was renamed ;)

File: docs/migration/3_0.md [hunk 2]
```diff
@@ -186,25 +186,35 @@ Now it is:
...
 
 - `getProduct` method will now only emit distinct product.
 
+### MultiCartStatePersistenceService
+
+- `sync` method has been renamed to `initSync`.
+
+### ProductReferenceService
+
+- `get` method has been removed. Use `getProductReferences` and `loadProductReferences` methods instead.
```
Comment (marlass) [inline docs/migration/3_0.md:371]: Can we provide `method and properties` migration for this?

File: projects/schematics/src/migrations/3_0/constructor-deprecations/data/checkout.service.migration.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      importPath: SPARTACUS_CORE,
+    },
+  ],
+  removeParams: [
+    {
+      className: AUTH_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+  ],
+  addParams: [
```
Comment (marlass) [inline projects/schematics/src/migrations/3_0/constructor-deprecations/data/checkout.service.migration.ts:36]: To make this migration work we need to move `UserIdService` in the service to be the last constructor parameter.

File: projects/schematics/src/migrations/3_0/constructor-deprecations/data/user-consent.service.migration.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    {
+      className: STORE,
+      importPath: NGRX_STORE,
+    },
+    {
+      className: AUTH_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+  ],
+  addParams: [
```
Comment (marlass) [inline projects/schematics/src/migrations/3_0/constructor-deprecations/data/user-consent.service.migration.ts:25]: We need to move this param to be last one in constructor for migration to work.

File: docs/migration/3_0.md [hunk 5]
```diff
@@ -217,3 +227,13 @@ Now it is:
...
 - `MultiCartService` now requires the additional provider `UserIdService`.
 - `PageSlotComponent` no longer uses `CmsComponentsService`. This service usage was replaces with the `PageSlotService`.
+- `ForbiddenHandler` now uses `GlobalMessageService`, `AuthService`, `OccEndpointsService`.
+- `CheckoutPaymentService` no longer uses `AuthService`. This service usage was replaced with the corresponding methods from `UserIdService`.
+- `CheckoutService` no longer uses `AuthService`. This service usage was replaced with the corresponding methods from `UserIdService`.
+- `CustomerCouponService` no longer uses `AuthService`. This service usage was replaced with the corresponding methods from `UserIdService`.
+- `OrderReturnRequestService` no longer uses `AuthService`. This service usage was replaced with the corresponding methods from `UserIdService`.
+- `UserAddressService` no longer uses `AuthService`. This service usage was replaced with the corresponding methods from `UserIdService`.
+- `UserConsentService` no longer uses `AuthService`. This service usage was replaced with the corresponding methods from `UserIdService`.
```
Comment (marlass) [inline docs/migration/3_0.md]: It still requires `AuthService`. Only inform that UserIdService is also required there.

**Review by marlass — CHANGES_REQUESTED** [review]
Also `projects/core/src/global-message/http-interceptors/handlers/unauthorized/unauthorized.handler.ts - removed` this migration wasn't covered in this PR.

File: docs/migration/3_0.md (hunk 2 above)
Comment (psmul) [reply docs/migration/3_0.md:371]: It was already done in other PR, but no doc mention, so I decided to supply it

**Review by marlass — DISMISSED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**